driver-core/src/main/java/com/datastax/driver/core/CodecRegistry.java

```diff
@@ -404,6 +405,27 @@ public CodecRegistry register(Iterable<? extends TypeCodec<?>> codecs) {
     }
 
     /**
+     * Returns a {@link TypeCodec codec} that accepts the given Java type.
+     * <p/>
+     * This method takes an arbitrary {@link TypeToken Java type} and tries to locate a suitable codec for it.
+     * This method returns the first matching codec, regardless of its accepted CQL type.
+     * It should be reserved for situations where the target CQL type is not available or unknown.
+     * This method can be used, for example, to determine the {@link DataType CQL type} corresponding to a
+     * particular {@link TypeToken Java type}, in order to generate a corresponding {@link com.datastax.driver.core.schemabuilder.Create}
+     * statement.
+     * <p/>
+     * Codecs returned by this method are <em>NOT</em> cached (see the {@link CodecRegistry top-level documentation}
+     * of this class for more explanations about caching).
+     *
+     * @param javaType The {@link TypeToken Java type} the codec should accept; must not be {@code null}.
+     * @return A suitable codec.
+     * @throws CodecNotFoundException if a suitable codec cannot be found.
+     */
+    public <T> TypeCodec<T> codecFor(TypeToken<T> javaType) {
+        return findCodec(null, javaType);
```

Comment:
What about 
```java
@Override
        public int weigh(CacheKey key, TypeCodec<?> value) {
            return codecs.contains(value) ? 0 : weigh(value.getCqlType(), 0);
        }
```
since the cqlType of the TypeCodec itself is never null?